test/code/scxsystemlib/disk/scxlvmutils_test.cpp

```diff
@@ -1036,6 +1039,143 @@ class SCXLVMUtilsTest : public CPPUNIT_NS::TestFixture
 
         CPPUNIT_ASSERT_EQUAL(((size_t) 3), index);
     }
+
+    /**
+     * This class mocks SCXLVMUtilsDepends for testing GetDMSlaves.
+     * Given a map M of strings to vectors of strings, this class simulates the following directory structure
+     * for each entry (X, [Y1,Y2,..Yn]) of M:
+     * /sys/block/X
+     * /sys/block/X/dev  <-- contains one line "253:2"
+     * /sys/block/X/slaves
+     * /sys/block/X/slaves/Y1
+     * /sys/block/X/slaves/Y2
+     * ....
+     * /sys/block/X/slaves/Yn
+     */
+    class TestGetDMSlavesSCXLVMUtilDepends
+        : public SCXSystemLib::SCXLVMUtilsDepends
+    {
+    public:
+        std::map< std::wstring, std::vector<std::wstring> > & m_slaves;
+
+        TestGetDMSlavesSCXLVMUtilDepends(std::map< std::wstring, std::vector< std::wstring > > & slaves) : m_slaves(slaves) { }
+
+        virtual std::vector< SCXCoreLib::SCXFilePath > GetFileSystemEntries(
+            const SCXCoreLib::SCXFilePath               & path,
+            const SCXCoreLib::SCXDirectorySearchOptions  options)
+        {
+            std::wstring slavesExt = L"/slaves/";
+            std::wstring pathStr = path.Get().substr(0, path.Get().length() - slavesExt.length());
+            SCXCoreLib::SCXFilePath blockFile = pathStr;
+
+            // to avoid getting unused variable warning
+            (void) options;
```

Comment:
Prefer 1071-1072 to be at line 1067 (top of routine) to be clearer that the parameter is not used.